// @cloudbase/extension-ci-wxmp/tcb.js
async function get(options) {
    return await request('GET', options)
}

async function post(options) {
    return await request('POST', options)
}

async function put(options) {
    return await request('PUT', options)
}

async function request(method, options) {

    const res = await new Promise((resolve, reject) => {
        const req = {
            url: options.url,
            header: options.headers,
            data: options.body,
            method,
            success(result) {
                if (result && result.code) {
                    reject(result)
                } else {
                    resolve(result)
                }
            },
            fail(err) {
                reject(err)
            }
        }
        wx.request(req)
    })

    return res
}

async function callFunction(options) {
    return await wx.cloud.callFunction(options)
}

const tcb = {
    requestClient: {
        get,
        post,
        put
    },
    callFunction
}

module.exports = tcb
